### PR TITLE
refactor(node): simplify 'node is blocking chain' logic

### PR DESCRIPTION
**Pull request by sergio-mena**
Contributes to #3415

This is mainly refactoring to simplify `onlyValidatorIsUs` and `localNodeBlocksTheChain` (since the latter implies the former).
It is a follow-up of #3406 (this is the part of #3406 that doesn't need to be backported)

---

#### PR checklist

- ~[ ] Tests written/updated~
- ~[ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~
- ~[ ] Updated relevant documentation (`docs/` or `spec/`) and code comments~

### CODE REVIEW

File: types/validator_set.go [hunk 1]
```diff
@@ -360,6 +360,14 @@ func (vals *ValidatorSet) findProposer() *Validator {
 	return proposer
 }
 
+func (vals *ValidatorSet) ValidatorBlocksTheChain(address []byte) bool {
+	_, val := vals.GetByAddress(address)
+	if val == nil {
+		return false
+	}
+	return val.VotingPower > vals.TotalVotingPower()/3
```
Comment (melekes) [inline types/validator_set.go:368]:
```suggestion
	return val.VotingPower >= vals.TotalVotingPower()/3
```

?

**Review by melekes — APPROVED** [review]

File: types/validator_set.go (hunk 1 above)
Comment (cason) [reply types/validator_set.go:368]: We probably need a test unit for this :)

File: node/setup.go [hunk 2]
```diff
@@ -309,7 +300,7 @@ func createMempoolAndMempoolReactor(
 		reactor := mempl.NewReactor(
 			config.Mempool,
 			mp,
-			waitSync,
+			true,
```
Comment (cason) [inline node/setup.go:303]: I assume this is always true because block sync is mandatory?

**Review by cason — APPROVED** [review]
Legit.

**Comment by cason** [comment]
There is a test error that might be related to the changes:
`
 2024/09/02 07:58:15 failed to unmarshal response: error in json rpc client, with http response metadata: (Status: 200 OK, Protocol HTTP/1.1) : RPC error -32603 - Internal error: endpoint is closed while node is catching up`

File: node/setup.go (hunk 2 above)
Comment (sergio-mena) [reply node/setup.go]: Actually it was so mindless that it was wrong 🤦.
Lucky, a couple of unit tests failed, which allowed me to spot the problem and validate the fix.

**Comment by sergio-mena** [comment]
This was the first time in a long time that unit tests, rather than e2e, saved me from screwing up 😌

**Comment by sergio-mena** [comment]
Unit tests really saved me here (rather than the usual e2e)

**Pull request merged**